### PR TITLE
Cumulate nominal

### PR DESCRIPTION
**Pull request by dcolleoni**
Compute cumulated nominal

### CODE REVIEW

File: ifrs17/Import/2ImportScope-PresentValue.ipynb [hunk 1]
```diff
@@ -655,8 +702,16 @@
...
+                "\n}"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "public interface RaCurrent : IScope<ImportIdentity, ImportStorage>",
```
Comment (dtrzesniak) [inline ifrs17/Import/2ImportScope-PresentValue.ipynb:714]: not thinking too much, but do you need a RaNominal?

File: ifrs17/Import/2ImportScope-PresentValue.ipynb [hunk 2]
```diff
@@ -479,16 +495,15 @@
...
                 "\n                                        .WithApplicability<CashflowAocStep>(x => x.Identity.Id.AocType == AocTypes.CF)",
-                "\n                                        .WithApplicability<PresentValueWithInterestAccretionForCreditRisk>(x => x.Identity.Id.IsReinsurance && x.Identity.AmountType == AmountTypes.CDR && x.Identity.Id.AocType == AocTypes.IA)",
                 "\n                                        .WithApplicability<PresentValueWithInterestAccretion>(x => x.Identity.Id.AocType == AocTypes.IA)",
-                "\n                                        .WithApplicability<EmptyValuesAocStep>(x => new string[]{AocTypes.BOP, AocTypes.EA, AocTypes.AM, AocTypes.RCU}.Contains(x.Identity.Id.AocType) ) //add here combination CRU for At !CDR?",
+                "\n                                        .WithApplicability<EmptyValuesAocStep>(x => new string[]{AocTypes.BOP, AocTypes.EA, AocTypes.AM, AocTypes.RCU}.Contains(x.Identity.Id.AocType) ||",
```
Comment (dtrzesniak) [inline ifrs17/Import/2ImportScope-PresentValue.ipynb:505]: can you extract a static string[], so you don't create this all he time?

File: ifrs17/Import/2ImportScope-PresentValue.ipynb [hunk 3]
```diff
@@ -96,7 +96,9 @@
...
+                "\nCalculation of present values can be performed using Current or Locked-in rates (locked at the inception year of the data node). During calculation, this is controlled by the Economic Basis which is either set to C for Current rates or L for Locked-in rates. In addition, the Economic Basis N identifies absence of discounting, corresponding to Nominal values. In this case an empty array is returned for both Interest and Discount values. ",
+                "\n",
+                "\n<br> The factors used for discounting have the same granularity as the cash flow, i.e. monthly. The yield curves have yearly granularity, so the annual Interest factor is 1 + interest rate. The monthly Interest Interest and Discount factors are obtained from the annual factors such that the product of 12 months results in the annual factors, as follows:",
```
Comment (dtrzesniak) [inline ifrs17/Import/2ImportScope-PresentValue.ipynb:101]: maybe this should be latex?

File: ifrs17/Import/2ImportScope-PresentValue.ipynb [hunk 4]
```diff
@@ -307,9 +318,9 @@
                 "\n{",
                 "\n    [NotVisible]",
                 "\n    string EconomicBasis => GetContext();",
-                "\n    private double[] CurrentValues => GetScope<DiscountedCashflow>(Identity).Values;",
+                "\n    protected double[] CurrentValues => GetScope<DiscountedCashflow>(Identity).Values;",
```
Comment (dtrzesniak) [inline ifrs17/Import/2ImportScope-PresentValue.ipynb:321]: can this be private again? maybe i missed the usage in the inherited interface....

**Review by dtrzesniak — APPROVED** [review]

File: ifrs17/Import/2ImportScope-PresentValue.ipynb (hunk 3 above)
Comment (dcolleoni) [reply ifrs17/Import/2ImportScope-PresentValue.ipynb]: I have improved the text here. There is already some latex for the formulas. I did not add formulas for the dependency on EconomicBasis.

**Pull request merged**